partition/roles/sonic-config/README.md

```diff
@@ -0,0 +1,499 @@
+```
+
+### sonic_ip_masquerade
+
+This variable was removed.
+See [this mini-lab PR](https://github.com/metal-stack/mini-lab/pull/237).
+
+### sonic_config_action
+
+This variable is replaced by the boolean variable `sonic_config_reload_config` which defaults to `false`.
+See [below](#variables) for explanation.
+
+### sonic_ports, sonic_ports_default_speed, sonic_ports_default_mtu and sonic_ports_default_fec
+
+All these variables are now gathered in the dictionary `sonic_config_ports`.
+
+Change
+
+```yaml
+sonic_ports_default_speed: 100000
+sonic_ports_default_mtu: 9000
+sonic_ports_default_fec: none
+
+sonic_ports:
+  - fec: rs
+    ips:
+      - 10.0.0.1/32
+    mtu: 1420
+    name: Ethernet0
+    speed: 40000
+    vrf: Vrf45
+```
+
+to
+
+```yaml
+sonic_config_ports:
+  # New field to configure default value for auto negotiation.
+  default_autoneg: # on
+  default_fec: none
+  default_mtu: 9000
+  list:
+    # New field to configure auto negotiation per port.
+    - autoneg: # off
+      fec: rs
+      ips:
+        - 10.0.0.1/32
+      mtu: 1420
+      name: Ethernet0
+      speed: 40000
+      vrf: Vrf45
+```
+
+`sonic_ports_default_speed` was removed.
+The default speed for a port is now derived from its breakout configuration and only a valid alternative speed can override it.
+
+### sonic_frr_debug_options
+
+This variable was removed.
+
+### sonic_interconnects_default_peer_group and sonic_interconnects_default_bgp_timers
+
+These two variables were removed.
+
+### sonic_portchannels and sonic_portchannels_default_mtu
+
+As with `sonic_ports`, these variables are now combined in one dictionary `sonic_config_portchannels`.
+
+Change
+
+```yaml
+sonic_portchannels_default_mtu: 9000
+
+sonic_portchannels:
+  - fallback: true
+    members:
+      - Ethernet0
+      - Ethernet1
+    mtu: 9000
+    number: 01
+```
+
+to
+
+```yaml
+sonic_config_portchannels:
+  default_mtu: 9000
+  list:
+    - fallback: true
+      members:
+        - Ethernet0
+        - Ethernet1
+      mtu: 9000
+      number: 01
+```
+
+## Variables
+
+This section shows an overview of all available variables along with some explanations.
+
+```yaml
+# The autonomous system number of the router.
+sonic_config_asn: # 42
+
+# Ports for the underlay BGP sessions.
+sonic_config_bgp_ports:
+#   - Ethernet120
+#   - Ethernet124
+
+# Map of breakout configuration for ports.
+sonic_config_breakouts:
+#   Ethernet0: 4x25G
+#   Ethernet4: 1x100G[40G]
+
+# Image of the [config generator tool](https://github.com/metal-stack/sonic-configdb-utils).
+sonic_configdb_utils_image_name: # ghcr.io/metal-stack/sonic-configdb-utils
+
+# Image tag (should be `ec202111` for Edgecore SONiC versions 202111.x).
+sonic_configdb_utils_image_tag: # ec202111
+
+# How to handle routing configuration.
+# With split mode all routing configuration is delegated to FRR.
+# With unified mode the FRR config gets generated by SONiC from its ConfigDB.
+sonic_config_docker_routing_config_mode: # split
```

Comment:
It would be nice to see possible values for these kind of enums. From the starting text I assume that the second mode is `split-unified`?